draft-ietf-tls-tls13.md

```diff
@@ -4695,22 +4695,33 @@ concerned with:
   handshake with B and probably retry the request, leading to duplication on
   the server system as a whole.
 
-The first class of attack can be prevented by the mechanism described
-in this section.  Servers need not permit 0-RTT at all, but those
-which do SHOULD implement either the single-use tickets or
-ClientHello recording techniques described in the following two
-sections.
+The first class of attack can be prevented by sharing state in orter to provide
+a guarantee that the 0-RTT is only accepted once.  Servers SHOULD provide that
+level of replay safety, by implementing one of the methods described in this
+section or by equivalent means.  It is understood, however, that due to
+operational concerns not all deployments will maintain the state at that level.
+Therefore, in normal operation, clients will not know which, if any, of these
+mechanisms servers actually implement and hence MUST only send early data which
+they are deem safe to be replayed.
+
+In addition to the direct effects of replays, there is a class of attacks where
+even operations normally considered idempotent could be exploited by a large
+number of replays (timing attacks, resource limit exhaustion and others
+described in {{replay-0rtt}}).  Those can be mitigated by providing a guarantee
+that every 0-RTT payload can be replayed only a limited number of times.  The
+server MUST ensure that any instance of it (be it a machine, a thread or any
+other entity within the relevant serving infrastructure) would accept the same
+0-RTT handshake only once; this limits the number of replays to the number of
```

Comment:
This really is a very nitty nit but you write "only once" here, but then "at most once" a short while later. I'd use the latter "at most once" in both places for consistency and pedantic precision.